Sming/SmingCore/Platform/Station.cpp

```diff
@@ -49,12 +49,12 @@ bool StationClass::isEnabled()
 	return wifi_get_opmode() & STATION_MODE;
 }
 
-bool StationClass::config(String ssid, String password, bool autoConnectOnStartup /* = true*/, bool save /* = true */)
+bool StationClass::config(const char * ssid, const char * password, bool autoConnectOnStartup /* = true*/, bool save /* = true */)
 {
 	station_config config = {0};
 
-	if (ssid.length() >= sizeof(config.ssid)) return false;
-	if (password.length() >= sizeof(config.password)) return false;
+	if (sizeof(ssid) >= sizeof(config.ssid)) return false;
```

Comment:
sizeof(ssid) is used incorrectly here. sizeof(char ) = 4 on esp8266 always. Strings cant be easily converted to char..:(